_posts/2019-07-29-custom-domains-function-ingress.md

```diff
@@ -0,0 +1,235 @@
+
+```sh
+export OPENFAAS_PREFIX=alexellis2
+
+faas template pull https://github.com/alexellis/static-site-template
+faas new --lang static-site-nginx my-homepage --prefix ${OPENFAAS_PREFIX}
+```
+
+Set `OPENFAAS_PREFIX` with your own Docker Hub username, or private registry address and repo.
+
+Create the homepage:
+
+```sh
+echo "<html>Hello world</html>" > ./my-homepage/index.html
+```
+
+The following will build a container image, push it to the remote registry and then deploy it to OpenFaaS, resulting in a static-website.
+
+```sh
+faas-cli up -f my-homepage.yml
+```
+
+Your traditional URL will appear on:
+
+```sh
+faas-cli describe -f my-homepage.yml my-homepage
+
+Name:                my-homepage
+Status:              Ready
+Replicas:            1
+Available replicas:  0
+Invocations:         0
+Image:               alexellis2/my-homepage:latest
+Function process:    
+URL:                 http://206.189.242.89:8080/function/my-homepage
+Async URL:           http://206.189.242.89:8080/async-function/my-homepage
+Labels:              faas_function : my-homepage
+```
+
+We'll now create a DNS entry and then the custom hostname mapping and TLS certificate along with that.
+
+### Create a DNS A record for your sub-domain
+
+My testing domain is `myfaas.club` and it cost me a very minimal amount of money from [namecheap.com](https://namecheap.com/). I can create sub-domains for each function, or for any testing that I do. If you are familiar with DNS-management, then create an A record for the IP address the LoadBalancer created by Nginx.
+
+```
+kubectl get svc -n default
+
+kubectl get svc
+NAME                                         TYPE           CLUSTER-IP      EXTERNAL-IP       PORT(S)                      AGE
+nginxingress-nginx-ingress-controller        LoadBalancer   10.245.169.18   178.128.137.209   80:32126/TCP,443:30028/TCP   32d
+```
+
+You can see my external IP listed as `178.128.137.209`.
+
+As a DigitalOcean customer, I can now use the DigitalOcean UI or CLI to create a DNS A record for `178.128.137.209`. Here is an example using the `doctl` CLI:
+
+```sh
+doctl compute domain create my-homepage.myfaas.club --ip-address 178.128.137.209
+
+Domain                     TTL
+my-homepage.myfaas.club    0
+```
+
+### Map the Custom Domain with a `FunctionIngress`
+
+Now we will map the Custom Domain to the function using a `FunctionIngress` definition.
+
+Save the following in a YAML file and then apply with `kubectl apply -f my-homepage-fni.yaml`
+
+```yaml
+apiVersion: openfaas.com/v1alpha2
+kind: FunctionIngress
+metadata:
+  name: my-homepage-tls
+  namespace: openfaas
+spec:
+  domain: "my-homepage.myfaas.club"
+  function: "my-homepage"
+  ingressType: "nginx"
+  tls:
+    enabled: true
+    issuerRef:
+      name: "letsencrypt-prod"
+      kind: "Issuer"
+```
+
+I've called my FunctionIngress record `my-homepage-tls`.
```

Comment:
What does this mean? Perhaps explain what a "FunctionIngress record" is so that the name could be changed if desired. The way it is, I don't know what that name is for, so not sure what could be a better name.